test/functional/test_framework/socks5.py

```diff
@@ -116,6 +127,30 @@ def handle(self):
             cmdin = Socks5Command(cmd, atyp, addr, port, username, password)
             self.serv.queue.put(cmdin)
             logger.info('Proxy: %s', cmdin)
+
+            if self.serv.conf.destinations_used < len(self.serv.conf.destinations):
```

Comment:
Does this guy need an `else`? Maybe raise error that a connection request was made after all destinations have been used?

Reply:
I didn't do that because my idea was to fall back to the old behavior of just closing the connection after the first `N` connections have been redirected (all destinations have been used). But you are right, I added a warning message now to ease diagnosing test behavior if that happens. Didn't `raise` because there is no error from the point of view of `socks5.py`.